instrumentation/net/http/httptrace/otelhttptrace/httptrace.go

```diff
@@ -39,7 +39,8 @@ func (o optionFunc) apply(c *config) {
 }
 
 type config struct {
-	propagators propagation.TextMapPropagator
+	tracerProvider 	trace.TracerProvider
+	propagators 	propagation.TextMapPropagator
```

Comment:
This actually isn't needed here.  This package is a bit messy and we probably should remove this part of it entirely since it isn't used many (any?) places that I'm aware of. 

The part that needs a `TracerProvider` option is `clienttrace.go` where it creates a `Tracer` from `otel.GetTracerProvider()`.

Reply:
Just making sure I'm understanding correctly, I would need to move `tracerProvider trace.TracerProvider` to `clienttrace.go`'s `clientTracer` struct, move `WithTracerProvider()` to `clienttrace.go`, and use that instead of `otel.GetTracerProvider()` to create a `Tracer`?

Reply:
After looking into it further, it seems that we still use `otel.GetTracerProvider()` to create a `Tracer`, but `WithTracerProvider()` is now available as a `ClientTraceOption`.